### PR TITLE
Fixing postcss not exiting with the proper exit code in --watch

### PR DESCRIPTION
**Pull request by GreenGremlin**
When running `postcss --watch`, if an error is encountered during the first pass postcss currently exits with an exit code of 0. The reason this is happening is because the error function specifically doesn't call `process.exit(1)` in watch mode.

### CODE REVIEW

**Comment by RyanZim** [comment]
`error()` is hard-coded to `process.exit(1)`: https://github.com/GreenGremlin/postcss-cli/blob/master/index.js#L294

**Comment by GreenGremlin** [comment]
Except for this `if (argv.watch) return`. In watch mode it returns before calling `process.exit(1)`.

**Comment by michael-ciniawsky** [comment]
In case you got a `SyntaxError` `postcss-cli`doesn't exit watch mode https://github.com/GreenGremlin/postcss-cli/blob/master/index.js#L290

**Comment by michael-ciniawsky** [comment]
Yep exactly

**Comment by michael-ciniawsky** [comment]
The CLI should recover from `SyntaxErrors` as these errors tend to happen during development, while a CLI/ Config Error normally is `FATAL` => `process.exit(1)`

**Comment by GreenGremlin** [comment]
A better solution would be to resolve with the list of files and not exit, but this is an easier change, for now.

**Comment by RyanZim** [comment]
Duh, yeah forgot about that line. That's totally intended; not changing this.

**Comment by GreenGremlin** [comment]
Can you explain why? This is an issue for me. I have a script that is running babel and postcss, both in watch mode, and I need to know if postcss exits with an error.

**Comment by michael-ciniawsky** [comment]
As mentioned, your solution would by a undesired breaking change since it breaks watch mode on `SyntaxError` and simply isn't justifiable to implement in the scope of this project for suiting **your** particular use case. If you wouldn't insists on letting `babel` handle File I/O for `postcss-cli` you will see the benefit of the current logic at last when accidentally making a typo will editing some CSS during development in watch mode yourself

**Comment by GreenGremlin** [comment]
My change doesn't break watch mode, as I'm only calling `process.exit(1)` when an error is encountered during the first pass. Watch mode already exits when there's a syntax error during the first run, but it is exiting with an exit code of 0.

Also, babel isn't handling file I/O, it just happens to be running at the same time as postcss. With my change postcss does not exit if you accidentally make a typo.

**Comment by RyanZim** [comment]
Ah, sorry, misunderstood your change. Yeah, for the first pass, that's debatable. Reopening for discussion.

File: index.js [hunk 1]
```diff
@@ -114,7 +114,10 @@ Promise.resolve()
         })
     }
   })
-  .catch(error)
+  .catch(err => {
+    error(err)
```
Comment (michael-ciniawsky) [inline index.js]: `\n`

Comment (GreenGremlin) [reply index.js]: done

**Comment by RyanZim** [comment]
Merged into `develop`, will be in the next major.

**Comment by RyanZim** [comment]
Released in v5, sorry for the long wait here.

**Pull request merged**